### ISSUE TITLE
关于数据集中joints标记的疑问

### ISSUE THREAD

**Issue by LucasForeverForward** [AlanJiang98/EvHandPose#6]
您好，非常感谢您的卓越贡献！在使用数据集时，我发现利用gt中标记的mano相关参数通过manopth得到的关节坐标，与gt中标记的关节坐标存在一定差距。请问这可能是因为什么原因？是我忽略了一些预处理么~
感激不尽。

**Comment by AlanJiang98** [comment]
MANO是对手的参数化表示，我们的mano是通过fitting到GT joints的方式得到的，经过各种尝试，最终存在平均0.5cm左右的误差。这在某种程度上反映了MANO对手表达能力的局限。

**Comment by LucasForeverForward** [comment]
在对模型进行测试的时候，我通过以下两种方式得到关节坐标：
gt_mano = annot['mano_hand_pose'][:,-1,:]
 gt_rot = annot['mano_rot_pose'][:,-1,:]
gt_trans = annot['mano_trans'][:,-1,:]
gt_shape = annot['mano_shape'][:,-1,:]
gt_joints = annot['joint'][:,-1,:,:] * 1000  #直接从标记中读取
gt_pose = torch.cat((gt_rot,gt_mano),dim=1)
#根据gt_mano gt_rot gt_trans gt_shape使用manolayer得到关节坐标
gt_hand_verts, gt_joints_3d = model.manolayer(
                    th_pose_coeffs=gt_pose,
                    th_betas=gt_shape,
                    th_trans=gt_trans)  #这里我使用的是manopth而非smplx
aligned_gt_joints = gt_joints - gt_joints[:,0:1,:]
aligned_gt_joints_3d = gt_joints_3d - gt_joints_3d[:,0:1,:]
print(aligned_gt_joints - aligned_gt_joints_3d)
最终print出的结果为：
tensor([[[  0.0000,   0.0000,   0.0000],
         [-19.0674, -63.4010, -15.1704],
         [-33.5518, -73.9206, -29.9962],
         [-43.9931, -71.1936, -37.6854],
         [-94.6013,  10.4785, -10.5457],
         [-35.9357, -31.2127, -23.9562],
         [-37.3180, -18.7257, -35.3418],
         [-49.1236,  90.4157,  -7.3652],
         [-66.7765,  95.1884, -12.8827],
         [-49.4541,   4.5395, -33.7822],
         [ -8.6802,  49.5266,   3.5048],
         [-18.5308,  43.0696,  -2.2135],
         [-13.0164,  50.0668, -11.9336],
         [ 54.3054,  56.7641,  34.7032],
         [ 79.8026,  61.7380,  31.1072],
         [100.2793,  66.7358,  23.2009],
         [126.7516,  76.4786,  16.2808],
         [ 62.5438, -95.0780, -18.3630],
         [ 26.1033, -86.7619, -21.4330],
         [ 13.6364, -53.6378, -14.5513],
         [  8.3228,  -3.1431,   1.3365]]], device='cuda:0')
可能暗示出二者之间并未对齐。请问是我对数据的理解存在一些问题吗~非常抱歉一直打扰您

**Comment by AlanJiang98** [comment]
值得check的地方：
1、https://github.com/AlanJiang98/EvHandPose/blob/main/scripts/eval.py#L108 使用SMPLX中MANO实现，参数设定，比如use_pca, is_rhand等；此外，我印象中MANO和manopth中一些计算有差别，我这边使用的时SMPLX，如果有兴趣可以对比下源码
2、https://github.com/AlanJiang98/EvHandPose/blob/main/core/dataset/dataset.py#L724 transform到相机坐标系时，MANO表达中的global trans并不是可以直接计算的，有一个translation的offset

**Comment by LucasForeverForward** [comment]
非常感谢您的回复~

**Comment by AlanJiang98** [comment]
https://github.com/hassony2/manopth/blob/master/manopth/manolayer.py#L260
我看到输出结果只有最后一行对的比较齐，可能是这行代码的原因

**Comment by LucasForeverForward** [comment]
是的，使用smplx就已经得到有效对齐了。非常感谢您的回复~

**Comment by M-leng** [comment]
> MANO是对手的参数化表示，我们的mano是通过fitting到GT joints的方式得到的，经过各种尝试，最终存在平均0.5cm左右的误差。这在某种程度上反映了MANO对手表达能力的局限。

您好，请问如何从Joints拟合到mano参数呀？
